### ISSUE TITLE
Support overriding tty detection (forcing interactive mode)

### ISSUE THREAD

**Issue by gibfahn** [dandavison/delta#101]
Simplest repro: `GIT_PAGER=delta git show | cat` gives the same output as `GIT_PAGER=cat git show` as it detects that the output is not a tty.

In `bat` you can set `--decorations=always --color=always` to force color and decorations to be output.

Refs: https://github.com/sharkdp/bat/issues/539 and https://github.com/sharkdp/bat/issues/448

This is useful for scripts, for example I have a more complicated version of this in my git config:

```ini
  lp = "!git log --graph --color=always --format=\"%C(auto)%h%d %s %C(black)%C(bold)<%an> %cr%C(auto)\" \"$@\" \
        | fzf --ansi -m --no-sort --reverse --tiebreak=index \
        --preview \"(grep -o '[a-f0-9]\\{7\\}' | head -1 | xargs -I % sh -c 'git -c core.pager=delta show --color=always %') <<< {}\" #"
```

I would like to be able to force delta to show colour and page in the preview window.

### Workaround

This seems to work (although I'm not entirely sure why):

```ini
  lp = "!git log --graph --color=always --format=\"%C(auto)%h%d %s %C(black)%C(bold)<%an> %cr%C(auto)\" \"$@\" \
        | fzf --ansi -m --no-sort --reverse --tiebreak=index \
        --preview \"(grep -o '[a-f0-9]\\{7\\}' | head -1 | xargs -I % sh -c 'git -c core.pager=cat show --color=never % | delta') <<< {}\" #"
```

it shows color, shows the decorations, and pages when you scroll in the preview window.

Basically it changes:

```diff
-GIT_PAGER=delta git show | cat
+GIT_PAGER=cat git show | delta | cat
```

when what I really want is:

```shell
GIT_PAGER="delta --decorations=always --color=always --paging=always" git show | cat
```

**Comment by dandavison** [comment]
Hi @gibfahn, this is some pretty cool fzf-delta hacking!

So firstly -- and I don't think it completely solves things here -- but delta (master branch) now has a `--paging=always|never|auto` option (added in https://github.com/dandavison/delta/pull/109).

I'm happy to help here. Could you help me understand exactly what more is needed? I might be misunderstanding various things, including the fzf preview tricks, but here's one thing I am thinking:

**If the output of the git process is not a tty then git just completely ignores its `core.pager` setting.**
 I.e. if we do `GIT_PAGER=__does_not_exist__ git show` then the first line of output is on stderr and is

```
error: cannot run __does_not_exist__: No such file or directory
```

However `GIT_PAGER=__does_not_exist__ git show > /dev/null` exits with code 0 and no output.

So,

> ...
> GIT_PAGER=cat git show | delta | cat
> when what I really want is:
> GIT_PAGER="delta --decorations=always --color=always --paging=always" git show | cat

If the `fzf` trick requires piping git somewhere, I believe we are _forced_ to pipe git into delta, rather than configuring git to use delta as its pager.

**Also, you probably know this, but delta _does_ currently output all decorations and color escape codes, even when its output is a pipe** (as one can see with `git show | delta --dark | cat -A`)

**And it also sends its output to less, even when its output is not going to a tty** (it always has done, which was perhaps a bug, so that might change, but now with `--paging=always` it will always do it and it will be a feature.) To prove this to myself I did

```
 ~/src/delta  (master) cat > ~/bin/less
echo "hello from fake less" 1>&2
cat -
 ~/src/delta  (master) chmod +x ~/bin/less
 ~/src/delta  (master) git show | delta --dark > /dev/null
hello from fake less
```

OK, I think there's more to discuss, and I'm seeing some things I don't fully understand with that last example... but hopefully this helps start the conversation!

**Comment by gibfahn** [comment]
>If the output of the git process is not a tty then git just completely ignores its core.pager setting.

Ahh, interesting, so I guess I misunderstood the issue, I can reproduce that, and the preview window definitely doesn't output to a tty.

---

So let me state the core issue, basically the added/deleted line highlighting runs on too far when in preview mode.

So to debug further I have the following in my git aliases:

```ini
  # Interactive log of $@.
  li = "!git log --graph --color=always --format=\"%C(auto)%h%d %s %C(black)%C(bold)<%an> %cr%C(auto)\" \"$@\" \
    | fzf --ansi -m --no-sort --reverse --tiebreak=index \
    --preview \"$preview\" \
    | grep -oE '[a-f0-9]{7,}' #"
```

That way you can set the preview command easily by passing `$preview`.

I ran a bunch of commands, the output in asciinema is the same as what I see in my terminal.

Some things I tried (see the asciicast):
- Latest master: overruns even more
- `--width=variable`: doesn't fix the issue
- Redirecting to a file: output is the same for `git show` and in the preview window.

[![asciicast](https://asciinema.org/a/306494.svg)](https://asciinema.org/a/306494)

<details><summary>Commands run:</summary>

```shell
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | ./target/release/delta' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta --width=variable --theme=none' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta --width=variable' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta | cat -A >~/tmp/delta-output' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta | cat -A' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta | cat' g li
preview='git show $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1) | delta' g li
preview='git show --color=always $(grep -oE "[a-f0-9]{7,}" <<< {} | head -1)' g li
```

</details>

<details><summary>Raw output:</summary>

commit 9418424182bf768aad19847716e17fdebb4471b8$
Author:     Dan Davison <dandavison7@gmail.com>$
AuthorDate: Sun Mar 1 21:35:48 2020 -0600$
Commit:     Dan Davison <dandavison7@gmail.com>$
CommitDate: Sun Mar 1 21:35:48 2020 -0600$
$
    Change a function name$
$
$
^[[34msrc/config.rs^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^P^[[0m$
^[[34m^[[38;2;171;178;191m ^[[38;2;198;120;221mpub^[[38;2;171;178;191m ^[[38;2;198;120;221mfn^[[38;2;171;178;191m ^[[38;2;97;175;239mget_config^[[38;2;171;178;191m<^[[38;2;198;120;221m'a^[[38;2;171;178;191m>(^[[0m ^[[34mM-bM-^TM-^B^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^X^[[0m$
^[[34m55^[[0m$
^[[38;2;171;178;191m     };                                                                       $
^[[38;2;171;178;191m                                                                              $
^[[38;2;171;178;191m     let minus_style_modifier = StyleModifier {                               $
^[[48;2;63;0;1m         background: Some(color_from_arg(^[[48;2;63;0;1m                                     $
^[[48;2;0;40;0m^[[38;2;171;178;191m         ^[[48;2;0;40;0m^[[38;2;224;108;117mbackground^[[48;2;0;40;0m^[[38;2;171;178;191m: Some(^[[48;2;0;40;0m^[[38;2;224;108;117mcolor_from_rgb_or_ansi_code_with_default^[[48;2;0;40;0m^[[38;2;171;178;191m(^[[48;2;0;40;0m           $
^[[38;2;171;178;191m             ^[[38;2;224;108;117mopt^[[38;2;171;178;191m.^[[38;2;224;108;117mminus_color^[[38;2;171;178;191m.^[[38;2;224;108;117mas_ref^[[38;2;171;178;191m(),                                        $
^[[38;2;171;178;191m             style::get_minus_color_default(^[[38;2;224;108;117mis_light_mode^[[38;2;171;178;191m, ^[[38;2;224;108;117mtrue_color^[[38;2;171;178;191m),       $
^[[38;2;171;178;191m         )),                                                                  $
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^P^[[0m$
^[[34m^[[38;2;171;178;191m ^[[38;2;198;120;221mpub^[[38;2;171;178;191m ^[[38;2;198;120;221mfn^[[38;2;171;178;191m ^[[38;2;97;175;239mget_config^[[38;2;171;178;191m<^[[38;2;198;120;221m'a^[[38;2;171;178;191m>(^[[0m ^[[34mM-bM-^TM-^B^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^X^[[0m$
^[[34m68^[[0m$
^[[38;2;171;178;191m     };                                                                       $
^[[38;2;171;178;191m                                                                              $
^[[38;2;171;178;191m     let minus_emph_style_modifier = StyleModifier {                          $
^[[48;2;63;0;1m         background: Some(color_from_arg(^[[48;2;63;0;1m                                     $
^[[48;2;0;40;0m^[[38;2;171;178;191m         ^[[48;2;0;40;0m^[[38;2;224;108;117mbackground^[[48;2;0;40;0m^[[38;2;171;178;191m: Some(^[[48;2;0;40;0m^[[38;2;224;108;117mcolor_from_rgb_or_ansi_code_with_default^[[48;2;0;40;0m^[[38;2;171;178;191m(^[[48;2;0;40;0m           $
^[[38;2;171;178;191m             ^[[38;2;224;108;117mopt^[[38;2;171;178;191m.^[[38;2;224;108;117mminus_emph_color^[[38;2;171;178;191m.^[[38;2;224;108;117mas_ref^[[38;2;171;178;191m(),                                   $
^[[38;2;171;178;191m             style::get_minus_emph_color_default(^[[38;2;224;108;117mis_light_mode^[[38;2;171;178;191m, ^[[38;2;224;108;117mtrue_color^[[38;2;171;178;191m),  $
^[[38;2;171;178;191m         )),                                                                  $
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^P^[[0m$
^[[34m^[[38;2;171;178;191m ^[[38;2;198;120;221mpub^[[38;2;171;178;191m ^[[38;2;198;120;221mfn^[[38;2;171;178;191m ^[[38;2;97;175;239mget_config^[[38;2;171;178;191m<^[[38;2;198;120;221m'a^[[38;2;171;178;191m>(^[[0m ^[[34mM-bM-^TM-^B^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^X^[[0m$
^[[34m81^[[0m$
^[[38;2;171;178;191m     };                                                                       $
^[[38;2;171;178;191m                                                                              $
^[[38;2;171;178;191m     let plus_style_modifier = StyleModifier {                                $
^[[48;2;63;0;1m         background: Some(color_from_arg(^[[48;2;63;0;1m                                     $
^[[48;2;0;40;0m^[[38;2;171;178;191m         ^[[48;2;0;40;0m^[[38;2;224;108;117mbackground^[[48;2;0;40;0m^[[38;2;171;178;191m: Some(^[[48;2;0;40;0m^[[38;2;224;108;117mcolor_from_rgb_or_ansi_code_with_default^[[48;2;0;40;0m^[[38;2;171;178;191m(^[[48;2;0;40;0m           $
^[[38;2;171;178;191m             ^[[38;2;224;108;117mopt^[[38;2;171;178;191m.^[[38;2;224;108;117mplus_color^[[38;2;171;178;191m.^[[38;2;224;108;117mas_ref^[[38;2;171;178;191m(),                                         $
^[[38;2;171;178;191m             style::get_plus_color_default(^[[38;2;224;108;117mis_light_mode^[[38;2;171;178;191m, ^[[38;2;224;108;117mtrue_color^[[38;2;171;178;191m),        $
^[[38;2;171;178;191m         )),                                                                  $
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^P^[[0m$
^[[34m^[[38;2;171;178;191m ^[[38;2;198;120;221mpub^[[38;2;171;178;191m ^[[38;2;198;120;221mfn^[[38;2;171;178;191m ^[[38;2;97;175;239mget_config^[[38;2;171;178;191m<^[[38;2;198;120;221m'a^[[38;2;171;178;191m>(^[[0m ^[[34mM-bM-^TM-^B^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^X^[[0m$
^[[34m90^[[0m$
^[[38;2;171;178;191m     };                                                                       $
^[[38;2;171;178;191m                                                                              $
^[[38;2;171;178;191m     let plus_emph_style_modifier = StyleModifier {                           $
^[[48;2;63;0;1m         background: Some(color_from_arg(^[[48;2;63;0;1m                                     $
^[[48;2;0;40;0m^[[38;2;171;178;191m         ^[[48;2;0;40;0m^[[38;2;224;108;117mbackground^[[48;2;0;40;0m^[[38;2;171;178;191m: Some(^[[48;2;0;40;0m^[[38;2;224;108;117mcolor_from_rgb_or_ansi_code_with_default^[[48;2;0;40;0m^[[38;2;171;178;191m(^[[48;2;0;40;0m           $
^[[38;2;171;178;191m             ^[[38;2;224;108;117mopt^[[38;2;171;178;191m.^[[38;2;224;108;117mplus_emph_color^[[38;2;171;178;191m.^[[38;2;224;108;117mas_ref^[[38;2;171;178;191m(),                                    $
^[[38;2;171;178;191m             style::get_plus_emph_color_default(^[[38;2;224;108;117mis_light_mode^[[38;2;171;178;191m, ^[[38;2;224;108;117mtrue_color^[[38;2;171;178;191m),   $
^[[38;2;171;178;191m         )),                                                                  $
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^P^[[0m$
^[[34m^[[38;2;171;178;191m ^[[38;2;198;120;221mfn^[[38;2;171;178;191m ^[[38;2;97;175;239mcolor_from_rgb_or_ansi_code^[[38;2;171;178;191m(^[[38;2;224;108;117ms^[[38;2;171;178;191m: &^[[38;2;198;120;221mstr^[[38;2;171;178;191m) -> Color {^[[0m ^[[34mM-bM-^TM-^B^[[0m$
^[[34mM-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@M-bM-^TM-^@^[[0m^[[34mM-bM-^TM-^X^[[0m$
^[[34m187^[[0m$
^[[38;2;171;178;191m     }                                                                        $
^[[38;2;171;178;191m }                                                                            $
^[[38;2;171;178;191m                                                                              $
^[[48;2;63;0;1m fn ^[[48;2;144;16;17mcolor_from_arg^[[48;2;63;0;1m(arg: Option<&String>, default: Color) -> Color {^[[48;2;63;0;1m           $
^[[48;2;0;40;0m^[[38;2;171;178;191m ^[[48;2;0;40;0m^[[38;2;198;120;221mfn^[[48;2;0;40;0m^[[38;2;171;178;191m ^[[48;2;0;96;0m^[[38;2;97;175;239mcolor_from_rgb_or_ansi_code_with_default^[[48;2;0;40;0m^[[38;2;171;178;191m(^[[48;2;0;40;0m^[[38;2;224;108;117marg^[[48;2;0;40;0m^[[38;2;171;178;191m: Option<&String>, ^[[48;2;0;40;0m^[[38;2;224;108;117mdefault^[[48;2;0;40;0m^[[38;2;171;178;191m: Color) -> Color {$
^[[38;2;171;178;191m     ^[[38;2;198;120;221mmatch^[[38;2;171;178;191m arg {                                                              $
^[[38;2;171;178;191m         Some(string) => ^[[38;2;86;182;194mcolor_from_rgb_or_ansi_code^[[38;2;171;178;191m(&string),                $
^[[38;2;171;178;191m         None => default,                                                     $
```

</details>

**Comment by gibfahn** [comment]
I was thinking that maybe this is the "zsh changing COLUMNS" issue, but the ruler is the correct width, so I'm guessing it's not:

https://github.com/junegunn/fzf/blob/9f0626da643070455fb5d6bb11ff1f483a0c100f/man/man1/fzf.1#L334-L338

**Comment by dandavison** [comment]
> basically the added/deleted line highlighting runs on too far when in preview mode.

OK. There's a colors running-on bug reported here: https://github.com/dandavison/delta/issues/87#issuecomment-575463707 and also here https://github.com/jesseduffield/lazygit/pull/690#issuecomment-593276408. I'll look into these in the next day or two when I get a chance (cc @jesseduffield).

**Comment by dandavison** [comment]
@gibfahn I've merged a branch #114 that I am hoping will fix the color overrun bug you're seeing. If you get a chance to test out delta master branch that would be great.

**Comment by gibfahn** [comment]
Yeah, looks like it fixes things:

### `git show --color=always`

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/15943089/75889147-5e2c0c00-5e24-11ea-865e-3282e0efb2b7.png">

### delta 0.0.16

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/15943089/75889342-ad723c80-5e24-11ea-9a04-00d01c9eebc9.png">

### delta 0.0.17

<img width="1173" alt="image" src="https://user-images.githubusercontent.com/15943089/75889267-8b78ba00-5e24-11ea-9e90-583eba40e8e9.png">
